pkg/network/resolver.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package network
+
+import (
+	"fmt"
+	"net/netip"
+
+	"go4.org/intern"
+
+	"github.com/DataDog/datadog-agent/pkg/network/slice"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// LocalResolver resolves connection remote addresses
+type LocalResolver struct {
+	processEventsEnabled bool
+}
+
+// NewLocalResolver creates a new LocalResolver
+func NewLocalResolver(processEventsEnabled bool) LocalResolver {
+	return LocalResolver{
+		processEventsEnabled: processEventsEnabled,
+	}
+}
+
+// Resolve binds container IDs to the Raddr of connections
+//
+// An attempt is made to resolve as many local containers as possible.
+//
+// First, we go over all connections resolving the laddr container
+// using the pid to container map that we have. We also index
+// connections by (laddr, raddr, proto, netns), inserting a
+// non-loopback saddr with netns = 0 as well. Translated
+// laddr and raddr are used throughout.
+//
+// Second, we go over the connections again, this time resolving
+// the raddr container id using the lookup table we built previously.
+// Translated addresses are used throughout.
+//
+// Only connections that are local are resolved, i.e., for
+// which conn.IntrHost is set to true.
+func (r LocalResolver) Resolve(conns slice.Chain[ConnectionStats]) bool {
```

Comment:
Do we run a flavor of this in the process-agent, and if so, is there code we can share?

Reply:
Likely, main thing is the connection type is different between the two. We can leave this for a follow up.